vars/buildFailureMessage.groovy

```diff
@@ -0,0 +1,17 @@
+@NonCPS
+def call(){
+    def ERROR_STRING = "ERROR"
+    def message = ""
+    String delimiter = ","
+    def performance_log = currentBuild.rawBuild.getLog(Integer.MAX_VALUE).join('\n')
+    performance_log.eachLine() { line ->
```

Comment:
I'm wonder how long this normally takes since it's iterating through the log? This could be inefficient if the log is long. Could we possibly read through the bottom of the log where the error is generally close to?

Reply:
Hey @zelinh, `currentBuild.rawBuild.getLog(Integer.MAX_VALUE)` we can add a value in place of Integer.MAX_VALUE, but this will get only last few lines of that value, and there are high chances we miss the ERROR message in those lines, so which is why I have added all lines and go over each line in running log, also this is part of post() script, which runs as part of same pipeline, so it should go over all the currentBuild log to get that ERROR string.

Reply:
Got it. Thanks!

Reply:
yes right now this code reads line by line, and filters for the log error message, I'm looking for a buffered stream here, as explained above this is part of Same pipeline as post() call, so we need to read all lines each line as part of same pipeline.

Reply:
Yes, but it currently goes through all the lines twice and makes 2 strings. Can we change it to only once?